### PR TITLE
FIO tests saved in perf_2 of influxdb

### PR DESCRIPTION
**Pull request by m-dati**
FIO storage tests in publiccloud, updated to write data into the new bucket `perf_2` of Influxdb

Some parametrs added to the db/record structure, to better identify the tested storage: DISTRI, ARCH, REGION, jobID, PUBLIC_CLOUD_PROVIDER.

`PUBLIC_CLOUD_FIO_TABLE` can be used to send data to a different table of the DB; default table is `storage_fio`.

- Related ticket: PR[152035](https://progress.opensuse.org/issues/152035)
- Needles:none
- Verification run: 
  - sle-15-SP6-EC2-BYOS-x86_64-Build0457-publiccloud_fio.VR14 @ec2_t3.small
    https://openqa.suse.de/tests/13072766#

The data produced in above VR have been also checked on Influxdb in the assigned table and are all ok as expected.

### CODE REVIEW

File: tests/publiccloud/storage_perf.pm [hunk 1]
```diff
@@ -161,11 +168,11 @@ sub run {
         # Store values in influx-db
         if ($url) {
             my $data = {
-                table => 'storage',
+                table => 'storage_1',
```
Comment (m-dati) [inline tests/publiccloud/storage_perf.pm:171]: In this first commit I named the table `storage_1`, to save verification data only and easier find those data, keeping separate the production table.
When PR will be ready to merge, the table will be named `storage_fio`.

**Comment by m-dati** [comment]
VR 2 more tests ok, with data pushed to Influxdb ok:

- sle-15-SP6-EC2-BYOS-x86_64-Build0457-publiccloud_fio@ec2_t3.small - 
  https://openqa.suse.de/tests/13072990

- sle-15-SP6-EC2-BYOS-x86_64-Build0457-publiccloud_fio@ec2_t3.small - 
  https://openqa.suse.de/tests/13073051

Please note that in these VRs I named the destination table `PUBLIC_CLOUD_FIO_TABLE=
storage_1`, to keep VR data only, and quick find those values on Influxdb.
When PR will be ready to merge, the default name `storage_fio` will be used, as production table.

File: tests/publiccloud/storage_perf.pm [hunk 2]
```diff
@@ -161,11 +168,11 @@ sub run {
         # Store values in influx-db
         if ($url) {
             my $data = {
-                table => 'storage',
+                table => get_var('PUBLIC_CLOUD_FIO_TABLE', 'storage_fio'),
```
Comment (asmorodskyi) [inline tests/publiccloud/storage_perf.pm:171]: I don't think this is something which suppose to be dynamic . Such things suppose to be hardlinked to certain test and I don't see why do you want to have possibility to have same test type to write into different tables

File: tests/publiccloud/storage_perf.pm [hunk 3]
```diff
@@ -103,9 +104,14 @@ sub run {
 
     my $tags = {
         instance_type => get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
+        job_id => get_current_job_id(),
+        os_provider => get_var('PUBLIC_CLOUD_PROVIDER'),
```
Comment (grisu48) [inline tests/publiccloud/storage_perf.pm:108]:
```suggestion
        os_provider => get_required_var('PUBLIC_CLOUD_PROVIDER'),
```
I think this tag is required to ensure that all collected measurements will have this tag.

File: tests/publiccloud/storage_perf.pm [hunk 4]
```diff
@@ -103,9 +104,14 @@ sub run {
...
+        os_build => get_required_var('BUILD'),
         os_flavor => get_required_var('FLAVOR'),
         os_version => get_required_var('VERSION'),
-        os_build => get_required_var('BUILD'),
+        os_distri => get_var('DISTRI'),
```
Comment (grisu48) [inline tests/publiccloud/storage_perf.pm:112]:
```suggestion
        os_distri => get_required_var('DISTRI'),
```

File: tests/publiccloud/storage_perf.pm [hunk 5]
```diff
@@ -103,9 +104,14 @@ sub run {
...
+        os_arch => get_var('ARCH'),
```
Comment (grisu48) [inline tests/publiccloud/storage_perf.pm:113]:
```suggestion
        os_arch => get_required_var('ARCH'),
```

**Review by grisu48 — COMMENTED** [review]
Some minor comments, overall ok

File: tests/publiccloud/storage_perf.pm (hunk 2 above)
Comment (m-dati) [reply tests/publiccloud/storage_perf.pm:171]: The idea comes noting that we already have parameters for db `PUBLIC_CLOUD_PERF_DB` and even url `PUBLIC_CLOUD_PERF_DB_URI`, rarely changing, but table don't, useful instead to address temporary VR tests data separated from production environment.

Comment (m-dati) [reply tests/publiccloud/storage_perf.pm:171]: Anyway, I will restore the hardlink

File: tests/publiccloud/storage_perf.pm (hunk 3 above)
Comment (m-dati) [reply tests/publiccloud/storage_perf.pm:108]: In this case also DB_URI will be required.

File: tests/publiccloud/storage_perf.pm (hunk 2 above)
Comment (m-dati) [reply tests/publiccloud/storage_perf.pm:171]: I set the table name: `storage_fio`

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: tests/publiccloud/storage_perf.pm [hunk 6]
```diff
@@ -67,11 +68,11 @@ sub analyze_previous_series {
 
 sub run {
     my ($self) = @_;
-    my $reg_code = get_var('SCC_REGCODE');
+
     my $runtime = get_var('PUBLIC_CLOUD_FIO_RUNTIME', 300);
     my $disk_size = get_var('PUBLIC_CLOUD_HDD2_SIZE');
     my $disk_type = get_var('PUBLIC_CLOUD_HDD2_TYPE');
-    my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
+    my $url = get_required_var('PUBLIC_CLOUD_PERF_DB_URI');
```
Comment (grisu48) [inline tests/publiccloud/storage_perf.pm:75]:
```suggestion
    my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
```
This variable was so far optional and should remain that way.

Comment (m-dati) [reply tests/publiccloud/storage_perf.pm]: ok, previous form restored

File: tests/publiccloud/storage_perf.pm (hunk 3 above)
Comment (m-dati) [reply tests/publiccloud/storage_perf.pm]: ok, this last not applicable.

**Pull request merged**